### DISCUSSION TITLE
why, where and how is invisible-watermark library used in the latest update of SD.Next?

### DISCUSSION THREAD

**Discussion by Sniper199999** [vladmandic/sdnext#1628] [General]
https://github.com/patrickvonplaten/invisible-watermark . It looks like it creates invisible watermark in images, but why would it be needed except classifying AI or original images?

**Comment by vladmandic** [comment]
When I add that as a feature, it will be exposed to users to customize.
For now, it's included only because it's implicit dependency of sd-xl model.

**Comment by Tillerz** [comment]
I wonder why when you switch to diffusers in the UI, it isn't automatically picking up the lib correctly for everyone but throwing an error for some, that it is not available, and you need to do a --reinstall to fix this. Is this just a thing that you need to figure out/fix yet?

**Reply by vladmandic** [reply]
installer was overly optimistic skipping checking requirements if it though nothing changed. i'll change it so it always checks requirements when using `--upgrade`.

**Reply by Tillerz** [reply]
Thanks! 

(I mean I fixed it via --reinstall, but every other person who has yet to upgrade will be glad!)

And I really like the output, at least it didn't get worse despite the default model being used and not some optimized ones by the community. This is a 1024x1024 render done in 7 seconds on an rtx 4090 (it has another broken hand though):

prompt: `beautiful twenty years old irish woman Mirte Bobina with auburn curls sitting in a cafe, sipping a tea.` - no negative prompt used, render steps: 40, euler a, CFG scale 6, CLIP skip 1.
![Screenshot1](https://i.gyazo.com/e44a4577371d2c08a1672e6228d0537f.jpg)

Same prompt and seed, but UniPC instead of euler a:

![Screenshot2](https://i.gyazo.com/80ded34af98fcb74ab2a00c0a52238dc.jpg)
